what4/src/What4/Protocol/SMTLib2.hs

```diff
@@ -702,10 +705,11 @@ parseRealSolverValue s = fail $ "Could not parse solver value: " ++ show s
 
 parseBvSolverValue :: MonadFail m => NatRepr w -> SExp -> m (BV.BV w)
 parseBvSolverValue w s
-  | Pair w' bv <- parseBVLitHelper s = case w' `testEquality` w of
-      Just Refl -> return bv
-      Nothing -> fail $ "Solver value parsed with width " ++
-                 show w' ++ ", but should have width " ++ show w
+  | Pair w' bv <- parseBVLitHelper s = case w' `compareNat` w of
+      NatLT zw -> return (BV.zext (addNat w' (addNat zw knownNat)) bv)
+      NatEQ -> return bv
+      NatGT _ -> fail $ "Solver value parsed with width " ++
```

Comment:
Should we use `panic` here instead?

Reply:
Good question. I want to look into how this error gets propagated a little more carefully. It can happy from a poorly-behaved external process, so I'd like to be able to recover gracefully.

Reply:
Yeah, this should probably be an exception instead, here and elsewhere.  But that should maybe be a separate PR.